src/animationBuilder.tsx

```diff
@@ -25,12 +26,52 @@ const mockTargetValues: LayoutAnimationsValues = {
   currentBorderRadius: 0,
 };
 
+function maybeReportOverwrittenProperties(
+  layoutAnimationStyle: StyleProps,
+  style: NestedArray<StyleProps>,
+  displayName: string
+) {
+  function getCommonProperties(
+    layoutStyle: object,
+    secondStyle: object | Array<object>
+  ) {
+    const secondStyleFlat = Array.isArray(secondStyle)
+      ? secondStyle.flat()
+      : [secondStyle];
+    const commonKeys: Array<string> = [];
+
+    secondStyleFlat.forEach((s) => {
+      if ('initial' in s) {
+        s = s.initial.value;
+      }
```

Comment:
Why do we need this piece of code?

Reply:
To include animated style, for example here we should throw an error: https://github.com/software-mansion/react-native-reanimated/issues/5395

Reply:
Okay, this makes sense, so `'initial' in s` is to check if `s` is an animated style?

If yes, maybe we should add `isAnimatedStyle` function in a separate PR for cases like this one.